MonMaraicher/Features/Map/MapView.swift

```diff
@@ -14,23 +14,32 @@ struct MapView: View {
 
     var body: some View {
         Map(position: $viewModel.mapCameraPosition, selection: $viewModel.selectedMarker) {
-
             ForEach(viewModel.allMarkers, id: \.id) { marker in
                 Marker(marker.title, systemImage: marker.systemImage, coordinate: marker.coordinate)
                     .tag(marker)
                     .tint(.orange)
             }
-
             UserAnnotation()
-
         }
         .mapStyle(.standard(elevation: .realistic))
         .mapControls {
             MapUserLocationButton()
         }
+        .overlay(alignment: .center) {
+            if viewModel.farmersLoadingInProgress && viewModel.hasUserAcceptedLocation {
+                loadingBar
+            }
+        }
+        .onChange(of: viewModel.hasUserAcceptedLocation ) {
+            viewModel.onReloadingFarmersButtonTapped()
```

Comment:
Something wrong here. You are calling a button action func. No one has tap on it right?

Reply:
In reality, it's worse than that. If the MapCamera does not change, then the action is not called and indeed, it is not an action I want but reload farmers

Reply:
I have not yet thought enough about how to observe the user's location changes

Reply:
Why not create a new method for that? That method could call the same code as the `onReloadingFarmersButtonTapped` and that's it.